src/snowcli/cli/nativeapp/introspection.py

```diff
@@ -0,0 +1,77 @@
+import re
+from pathlib import Path
+from typing import List, Union
+
+from click import ClickException
+from snowcli.cli.nativeapp.artifacts import resolve_without_follow, NotInDeployRootError
+from strictyaml import YAML, load
+
+EXECUTE_IMMEDIATE_FROM_REGEX = re.compile(
+    r"execute\s+immediate\s+from\s+'([^']+?)'", re.IGNORECASE
```

Comment:
could the file name include escaped single quote?

Reply:
Good idea. TIL that we also support block comments (`/* ... */`) so this will get pretty complicated quickly. I think the move is to pre-process out all comments and then only run the regex on the comment-less code.